### PR TITLE
Documentation updates for command line parameters and e2e tests

### PR DESCRIPTION
**Pull request by dmsuehir**
Command line doc updates for:
* noting that `--template-repo` can be a local path to a MLT repo.
* updating the `--retries` option description since it's used for both logging and `--interactive`.

Developer guide doc update to note that developers should use the `make test-e2e-no-docker` option and that `make test-e2e` is for circleci.

## Reviewer Checklist

 - [ ] Do you understand it?
 - [ ] Is it correct?
 - [ ] Is it safe?
 - [ ] Is it legally compliant?
 - [ ] Is it robust?
 - [ ] Is it simple?
 - [ ] Is it tested?
 - [ ] Is it documented?
 - [ ] Is the style consistent?

See [the reviewer guide](docs/reviews.md) for more detail on each check.

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -24,9 +24,11 @@ Then you can run `mlt` from anywhere and it'll run your current repository codeb
 
 ### E2E Testing Options
 
-`make test-e2e`: Use docker-compose to spin up a testing container and run tests; this uses hyperkube.
+`make test-e2e`: Use docker-compose to spin up a testing container and run tests.
+This is used by circleci to run tests on a Google Cloud test cluster.
 
 `make test-e2e-no-docker`: Use your local environment to run e2e tests, similar to the way `make test` runs unit tests.
+Developers should use this option.
```
Comment (ashahba) [inline docs/developer_guide.md]: Maybe like:
This is preferred way for developers for running e2e tests ??

Comment (ashahba) [reply docs/developer_guide.md]: But overall looks good.

Comment (dmsuehir) [reply docs/developer_guide.md]: It's more than just the preferred way.  The other way doesn't work locally (I tried it, and it failed and messed up my environment 😄).

Comment (s1113950) [reply docs/developer_guide.md]: Yeah @ashahba now it's the only way

**Pull request merged**